pkg/cloud/aws/actuators/scope.go

```diff
@@ -207,15 +209,28 @@ func (s *Scope) Close() {
 		s.Cluster.ResourceVersion = result.ResourceVersion
 	}
 
+	// Check if API endpoints is not set or has changed.
+	if s.Cluster.Status.APIEndpoints == nil || s.Cluster.Status.APIEndpoints[0].Host != s.ClusterStatus.Network.APIServerELB.DNSName {
```

Comment:
The problem can be summarized: 

The cluster status is only being updated if the cluster *ProviderStatus* has changed, but the status is more than *ProviderStatus*.

The main changes I'd like to see here are:

1. Instead of reflect.DeepEqual on the providerstatus, do a deep equal between `s.Cluster.Status` and `s.ClusterCopy.Status`. => delete the oldStatus var
2. Don't do an additional UpdateStatus call => remove the logger and the additional UpdateStatus call this PR adds
3. move line 230 above the logic this PR adds regarding the check for apiendpoints having changed
4. change the reflect.DeepEqual to compare the s.Cluster.Status and s.ClusterCopy.Status

Reply:
done